src/sidebar/services/tags.js

```diff
@@ -41,12 +41,15 @@ export class TagsService {
     //   (e.g. tag "pink banana" matches query "ban"), OR
     // * tag has substring query occurring after a non-word character
     //   (e.g. tag "pink!banana" matches query "ban")
-    let regex = new RegExp('(\\W|\\b)' + query, 'i');
     return savedTags.filter(tag => {
-      if (tag.match(regex)) {
-        if (limit === null || resultCount < limit) {
-          // limit allows a subset of the results
-          // See https://github.com/hypothesis/client/issues/1606
+      const words = tag.split(/\W+/);
+      let regex = new RegExp(`^${query}`, 'i'); // Only match the start of the string
```

Comment:
You need to escape the query when used within the regex. See `src/sidebar/helpers/groups.js` for an example of how to do this using the escape-string-regexp package. Also the regex should be `const` if not re-assigned.